packages/renderer/src/lib/preferences/PreferencesProviderRendering.svelte

```diff
@@ -50,14 +50,20 @@ async function stopProvider(): Promise<void> {
   window.dispatchEvent(new CustomEvent('provider-lifecycle-change'));
 }
 
-async function startReceivingLogs(provider: ProviderInfo): Promise<void> {
+async function startReceivingLogs(provider: ProviderInfo | undefined): Promise<void> {
+  if (!provider) {
```

Comment:
This is highly suspicious code that will cover up programming errors rather than helping to avoid them in a first place.

To avoid this check and similar in stopRecievingLogs method both methods should be called with `providerInfo` which cannot be `undefined` rater than with `showModal`.